src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBase.java

```diff
@@ -166,15 +139,17 @@ public GetRecordsResult getRecords(GetRecordsRequest request) {
             LOG.debug("getRecords request={}", request);
         }
 
-        ShardIterator iterator = ShardIterator.parse(request.getShardIterator());
+        ShardIterator iterator = ShardIterator.fromString(request.getShardIterator());
+        String arn = iterator.getArn();
+        StreamArn streamArn = parse(arn);
```

Comment:
If we're worried about performance we could memoize parsing of arn's.  But I'm generally in favor of not prematurely optimizing this especially given that I'm not even sure regionMatches is going to be a lot slower than doing it ourselves.  So, in conclusion, I think it's fine to merge as-is.